sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -98,6 +122,7 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
           val (newCond, inputPlan) = rewriteExistentialExpr(Seq(predicate), p)
           Project(p.output, Filter(newCond.get, inputPlan))
       }
+      dedupJoin(rewritten)
```

Comment:
add a comment above this line to explain it.

Reply:
After rethinking it, we can be more conservative. Instead of doing a dedup at the end, we should do it when we convert it to the `Join`.

Reply:
Fair point. Will follow it.